samza-sql-lib/src/main/java/org/apache/samza/sql/interfaces/SourceSchemaResolver.java

```diff
@@ -0,0 +1,36 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.interfaces;
+
+import java.util.Optional;
+
+
+/**
+ * Finds the schema associated with the source.
+ */
+public interface SourceSchemaResolver {
```

Comment:
Consider this scenario, right now at Linkedin we use avro for schema but we have multiple schema registries (Kafka, Espresso, HDFS). So we will have three implementations of AvroSchemaConverters each for different schema registry whereas it's just the resolvers that are really different. 

More over Schema resolvers and Schema converters are really different things conceptually needing different configs, quite different implementations. IMO it makes sense for them to be separate.

Reply:
Based on our discussion, I will move the SchemaResolver to be specific to AvroRelSchemaConverter, so that we will get pluggability around schema registry that we need.